### PR TITLE
Doc review

### PR DESCRIPTION
**Pull request by lionel-**

### CODE REVIEW

File: src/docs/objects.qmd [hunk 1]
```diff
@@ -8,47 +8,51 @@ toc: true
...
 
-Support for different types of R object are provided by different `RProxy` type aliases in the [`RMain` module](api/js/modules/RMain.md). Each type of `RObject` proxy is associated with a corresponding subclass of [`RWorker.RObject`](api/js/classes/RWorker.RObject.md) with the same name. The `RWorker` module implements the methods for interacting with R objects on the webR worker thread, and the methods are made available to the main thread through `RObject` proxies.
+A proxy is parameterised by an R type, e.g. a proxy that wrap a character vector is implemented as `RProxy<RWorker.RCharacter>`. The [`RMain` module](api/js/modules/RMain.md) provides convenient type aliases for each R type. For instance, `RCharacter` stands for `RProxy<RWorker.RCharacter>`. Each of these types supports specific methods related to the kind of data they represent. For instance `RCharacter` supports conversion to a JS string with `toString()` whereas `RDouble` supports `toNumber()`.
```
Comment (georgestagg) [inline src/docs/objects.qmd:19]:
```suggestion
A proxy is parameterised by an R type, e.g. a proxy that wraps a character vector is implemented as `RProxy<RWorker.RCharacter>`. The [`RMain` module](api/js/modules/RMain.md) provides convenient type aliases for each R type. For instance, `RCharacter` stands for `RProxy<RWorker.RCharacter>`. Each of these types supports specific methods related to the kind of data they represent. For instance `RCharacter` supports conversion to a JS string with `toString()` whereas `RDouble` supports `toNumber()`.
```

File: src/docs/objects.qmd [hunk 2]
```diff
@@ -60,26 +64,53 @@ try {
...
+
+try {
+  const obj1 = await myShelterR.evalR('42 + 123');
+  const obj2 = await webR.evalR('42 * 123');
+
+  const result = obj1.toNumber() + obj2.toNumber();
+  console.log(result);
+} finally {
+  myShelterR.purge();
+}
```
Comment (georgestagg) [inline src/docs/objects.qmd:109]:
```suggestion
const myShelter = await new webR.Shelter();

try {
  const obj1 = await myShelter.evalR('42 + 123');
  const obj2 = await webR.evalR('42 * 123');

  const result = await obj1.toNumber() + await obj2.toNumber();
  console.log(result);
} finally {
  myShelter.purge();
}
```

File: src/docs/objects.qmd [hunk 3]
```diff
@@ -60,26 +64,53 @@ try {
...
 
-A new shelter can be created using the [`Shelter`](api/js/classes/WebR.Shelter.md) class. An asynchronous `Shelter` constructor is available as an instance property of the `WebR` class. Creating a new object with this constructor returns a Promise that resolves to a new instance of the `Shelter` class once the shelter is ready to use.
+try {
+  let result = await obj1.toNumber();
+  const obj2 = await webR.evalR('42 * 123');
 
-``` javascript
-const myShelter = await new webR.Shelter();
+  try {
+    result = result + obj2.toNumber();
```
Comment (georgestagg) [inline src/docs/objects.qmd:77]:
```suggestion
    result = result + await obj2.toNumber();
```

**Review by georgestagg — APPROVED** [review]
Looks great, thanks! A few minor tweaks are below.

**Comment by georgestagg** [comment]
The section at the bottom of `objects.qmd` named "Destroying objects protected by a shelter" is probably now redundant with your new example of using `purge()` added in the previous section.

All it now adds is a link to `Shelter.destroy()`, so I suggest we remove that final section and simply add a bullet to the list of shelter features in the section above. Something like,

> * Individual objects protected by a shelter instance can be removed using `Shelter.destroy()`

**Comment by lionel-** [comment]
oops I missed your last comment, done in https://github.com/georgestagg/webR/commit/45486eef8b18c42e71dd7f50c3583d44271fa1f3

**Pull request merged**